### PR TITLE
add elasticsearch db.statement sanitization

### LINKED ISSUE #1545: Implement sensitive data sanitization for elasticsearch instrumentation
Aggregated by #1543

### PR DESCRIPTION
**Pull request by nemoshlag**
# Description

Added an optional query sanitizer to the elasticsearch instrumentation.
Usage
ElasticsearchInstrumentor().instrument(sanitize_query=True)

This will affect the DB_STATEMENT value to contain the original query or sanitized one.

Fixes #1545 
Following the specification discussion [here](https://github.com/open-telemetry/opentelemetry-specification/issues/3104)

## Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [x] Test A db.statement has been sanitized
- [x] Test B no side affects occurred on a query without a db.statement attribute 

# Does This PR Require a Core Repo Change?

- [x] No.

# Checklist:

- [x] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [x] Unit tests have been added
- [x] Documentation has been updated

### CODE REVIEW

**Review by avzis — COMMENTED** [review]
Maybe add a description of the param 'sanitize_query' in the init file (line 54)

**Comment by shalevr** [comment]
Can we use the same sanitization function for all of the db libraries?

File: instrumentation/opentelemetry-instrumentation-elasticsearch/src/opentelemetry/instrumentation/elasticsearch/__init__.py [hunk 1]
```diff
@@ -214,7 +223,10 @@ def wrapper(wrapped, _, args, kwargs):
                 if method:
                     attributes["elasticsearch.method"] = method
                 if body:
-                    attributes[SpanAttributes.DB_STATEMENT] = str(body)
+                    statement = str(body)
+                    if sanitize_query:
+                        statement = "sanitized"
```
[the remark this answers is not in the document]
Comment (srikanthccv) [reply instrumentation/opentelemetry-instrumentation-elasticsearch/src/opentelemetry/instrumentation/elasticsearch/__init__.py:228]: I agree with this. The statement work _sanitized_ is not the best representation of sanitized statement. It also good thing to keep consistency for statement sanitization.

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,7 +9,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
-- `opentelemetry-instrumentation-redis` Add `sanitize_query` config option to allow query sanitization.  ([#1572](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1572))
+- Add default query sanitization for elasticsearch db.statement attribute
+  ([#1545](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1545))
+- `opentelemetry-instrumentation-redis` Add `sanitize_query` config option to allow query sanitization.
```
Comment (shalevr) [inline CHANGELOG.md:14]: The merge with the master antecedently adds this entry

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,7 +9,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
-- `opentelemetry-instrumentation-redis` Add `sanitize_query` config option to allow query sanitization.  ([#1572](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1572))
+- Add default query sanitization for elasticsearch db.statement attribute
+  ([#1545](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1545))
```
Comment (shalevr) [inline CHANGELOG.md:13]: Your PR is not #1598 and
https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1598
?

**Comment by shalevr** [comment]
LGTM

**Review by shalevr — APPROVED** [review]

**Comment by phillipuniverse** [comment]
FYI this does not work with the elasticsearch bulk API, see #1868

**Pull request merged**